chore: cleanup organization of the process managers (on top of #416)

doudou opened the pull request:
Depends on:
- [ ] https://github.com/rock-core/tools-syskit-log/pull/49
- [ ] https://github.com/rock-gazebo/simulation-rock_gazebo/pull/55
- [ ] https://github.com/rock-gui/gui-orogen-qt_base/pull/2

On top of #416

The next PR will build on top of this to import Orocos::Process into syskit and clean that up.

doudou committed:
chore: cleanup organization of the process managers

They were all over the place, without any naming consistency.
Move them all to syskit/process_managers and call the manager
Manager and the process Process.

doudou committed:
chore: regen rubocop_todo.yml

wvmcastro commented on lib/syskit/process_managers/in_process/manager.rb:
```diff
@@ -0,0 +1,192 @@
+
+                # Requests that the process server moves the log directory at +log_dir+
+                # to +results_dir+
+                def save_log_dir(log_dir, results_dir); end
+
+                # Creates a new log dir, and save the given time tag in it (used later
+                # on by save_log_dir)
+                def create_log_dir(log_dir, time_tag, metadata = {}); end
```
should this raise `NotImplemented` ?

doudou replied:
No, it's a no-op. Will be removed later on once the process managers do not depend anymore on anything from `tools/orocos.rb`

wvmcastro commented on lib/syskit/process_managers/in_process/manager.rb:
```diff
@@ -0,0 +1,192 @@
+                #
+                # Returns a hash that maps deployment names to the Status
+                # object that represents their exit status.
+                def wait_termination(_timeout = nil)
```
The comment suggests that `_timeout` is useful but the implementation ignores it. Is it here just for some api consistence ?

doudou replied:
Yes. Consistence with the {Orocos::Process} API. The end goal is to remove all that old unused stuff by making all of this (1) syskit-specific and (2) private.

In Syskit, `timeout` is always zero.

wvmcastro commented on lib/syskit/process_managers/in_process/manager.rb:
```diff
@@ -0,0 +1,192 @@
+                    result = {}
+                    process_names.each do |name|
+                        if (p = processes[name])
+                            result[name] = { iors: p.wait_running(0) }
```
iors ?

doudou replied:
IOR is the string representation of an address in CORBA. It is what allows us to create Orocos::TaskContext to control orogen components.

doudou committed:
feat: implement a public API on Configuration to create and register process managers

This will allow plugins and other clients to be independent of how process
managers should be created, making changes easier moving forward.

doudou committed:
fix: race condition in Server#quit

doudou committed:
fix: remove the obsolete support for log_dir in remote server setup

doudou committed:
fix: do not pass working_directory from orocos.rb to syskit
